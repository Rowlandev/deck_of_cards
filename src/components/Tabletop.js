import React from "react";
import PlayingCard from "./PlayingCard.js";
import HelpButton from "./HelpButton.js";

class Tabletop extends React.Component {
  render() {
    return (
      <div>
        <HelpButton/>
        <PlayingCard suite="heart" value="5" color="red"/>
      </div>
    );
  }
}

export default Tabletop;
