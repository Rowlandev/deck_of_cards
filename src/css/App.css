html, body, #root {
  height: 100%;
  margin: 0px;
  padding: 0px;
}

.App {
  background: #76b852; /* fallback for old browsers */
  background: -webkit-linear-gradient(to right, #76b852, #8dc26f); /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(to right, #76b852, #8dc26f); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
  height: 100%;
}

#top-left-suite {
  margin: 0px;
  pading: 0px;
  background: red;
  width: auto;
}

#top-left-value {
  margin: 0px;
  pading: 0px;
  background: blue;
}

#bottom-right-suite {
  margin: 0px;
  pading: 0px;
  background: yellow;
}

#bottom-right-value {
  margin: 0px;
  pading: 0px;
  background: purple;
}
