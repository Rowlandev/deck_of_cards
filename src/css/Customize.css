#customize-menu {
  text-align: center;
  margin: 0;
}

#customize-title {
  margin: 0;
  text-align: center;
  font-family: "Andale Mono", AndaleMono, monospace;
  font-size: 50px;
  color: white;
  padding-top: 40px;
  padding-bottom: 40px;
}

#card-choices {
  border: solid white 1px;
  border-radius: 7px;
  background: black;
  margin: auto;
  width: 650px;
}

.card-backs {
  width: 143px;
  height: 200px;
  padding: 0;
  border-radius: 7%;
  border: solid gray 1px;
  margin: 0 auto;
  margin-left: 5px;
  margin-right: 5px;
  overflow: hidden;
}
