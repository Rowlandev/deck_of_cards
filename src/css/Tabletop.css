.main-menu {
  text-align: center;
  margin: 0;
}

#title {
  margin: 0;
  text-align: center;
  font-family: "Andale Mono", AndaleMono, monospace;
  font-size: 75px;
  color: white;
  padding-top: 40px;
}

#creators {
  font-family: "Andale Mono", AndaleMono, monospace;
  font-size: 20px;
  color: white;
  padding-bottom: 15px;
}

.button {
  background-color: white;
  width: 250px;
  border: solid black 1px;
  color: black;
  padding: 20px;
  font-family: "Andale Mono", AndaleMono, monospace;
  text-align: center;
  display: inline-block;
  font-size: 16px;
  border-radius: 7px;
  margin-top: 7px;
}

.button:hover {
  background-color: LightGray;
}

#settings-menu {
  text-align: center;
  margin: 0;
}

#settings-title {
  margin: 0;
  text-align: center;
  font-family: "Andale Mono", AndaleMono, monospace;
  font-size: 50px;
  color: white;
  padding-top: 40px;
  padding-bottom: 40px;
}

#card-choices {
  border: solid white 1px;
  border-radius: 7px;
  background: black;
  margin: auto;
  width: 650px;
}

.card-backs {
  width: 143px;
  height: 200px;
  padding: 0;
  border-radius: 7%;
  border: solid gray 1px;
  margin: 0 auto;
  margin-left: 5px;
  margin-right: 5px;
  overflow: hidden;
}
