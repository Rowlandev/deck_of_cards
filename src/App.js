import React from 'react';
import Tabletop from "./components/Tabletop.js";
import "./css/App.css";

function App() {
  return <div className="App"><Tabletop/></div>;
}

export default App;
