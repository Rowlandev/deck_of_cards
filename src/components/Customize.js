// SYSTEM IMPORTS
import React from "react";

import "../css/Customize.css";

// IMAGE IMPORTS
import Prof1 from "../images/backs/prof1.JPG"
import Prof2 from "../images/backs/prof2.PNG"
import Prof3 from "../images/backs/prof3.JPG"
import Tim from "../images/backs/tim.PNG"
import Flavortown from "../images/backs/flavortown.PNG"
import NickCage from "../images/backs/nick_cage.PNG"
import TheGuy from "../images/backs/the_guy.PNG"
import Squid from "../images/backs/squid.PNG"

class Customize extends React.Component {

  constructor(props) {
    super(props)
    this.state = {
    }
  }

  prof1Click = () => {
    this.props.changeImg(Prof1);
    this.props.goToMainMenu();
  }
  prof2Click = () => {
    this.props.changeImg(Prof2)
    this.props.goToMainMenu()
  }
  prof3Click = () => {
    this.props.changeImg(Prof3)
    this.props.goToMainMenu()
  }
  timClick = () => {
    this.props.changeImg(Tim)
    this.props.goToMainMenu()
  }
  flavortownClick = () => {
    this.props.changeImg(Flavortown)
    this.props.goToMainMenu()
  }
  theGuyClick = () => {
    this.props.changeImg(TheGuy)
    this.props.goToMainMenu()
  }
  nickCageClick = () => {
    this.props.changeImg(NickCage)
    this.props.goToMainMenu()
  }
  squidClick = () => {
    this.props.changeImg(Squid)
    this.props.goToMainMenu()
  }

  render() {
    return (
      <div id="customize-menu">
        <p id="customize-title">Choose your fighter...</p>
        <div id="card-choices">
          <p><img className="card-backs" alt="card back" src={Prof1} onClick={this.prof1Click}/>
             <img className="card-backs" alt="card back" src={Prof2} onClick={this.prof2Click}/>
             <img className="card-backs" alt="card back" src={Prof3} onClick={this.prof3Click}/>
             <img className="card-backs" alt="card back" src={Tim} onClick={this.timClick}/>
          </p>
          <p><img className="card-backs" alt="card back" src={Flavortown} onClick={this.flavortownClick}/>
             <img className="card-backs" alt="card back" src={TheGuy} onClick={this.theGuyClick}/>
             <img className="card-backs" alt="card back" src={NickCage} onClick={this.nickCageClick}/>
             <img className="card-backs" alt="card back" src={Squid} onClick={this.squidClick}/>
          </p>
          </div>
      </div>
    );
  }
}

export default Customize;
