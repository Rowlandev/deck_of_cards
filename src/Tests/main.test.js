// SYSTEM IMPORTS
import React from "react";

// TESTING TOOL IMPORTS
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';

// COMPONENT IMPORTS
import App from "../App.js";
import Tabletop from "../components/Tabletop.js";
import Customize from "../components/Customize.js";

describe('DECK OF CARD TESTS', () => {

  it('TEST #1 - APPLICATION RENDERS', () => {
    // create tree and confirm that it renders
    var tree = renderer.create(<App/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('TEST #2 - SHOWS MAIN PAGE', () => {
    //  Create wrapper & instance for Tabletop
    var wrapper = mount(<Tabletop/>);
    var instance = wrapper.instance();

    // Confirm app is showing initial main menu
    const result = instance.state.mode == "mainMenu" ? true : false;
    expect(result).toBe(true);
  });

  it('TEST #3 - RENDERS 3 MAIN PAGE OPTIONS', () => {
    //  Create wrapper & instance for Tabletop
    var wrapper = mount(<Tabletop/>);
    var instance = wrapper.instance();

    // Confirm app is showing initial main menu
    var result = instance.state.mode == "mainMenu" ? true : false;
    expect(result).toBe(true);

    // Confirm there are three page options rendered for the user
    const desiredNumberOfChoices = 3;
    const foundNumberOfChoices = wrapper.find('p').length - 2;
    result = desiredNumberOfChoices == foundNumberOfChoices ? true : false;
    expect(result).toBe(true);
  });

  it("TEST #4 - 3 MAIN PAGE OPTIONS LOAD CORRECLY ON CLICK", () => {
    //  Create wrapper & instance for Tabletop
    var wrapper = mount(<Tabletop/>);
    var instance = wrapper.instance();

    // Confirm app is showing initial main menu
    var result = instance.state.mode == "mainMenu" ? true : false;
    expect(result).toBe(true);

    // Change state to 'customize' to simulate click
    instance.state.mode = "customize";
    result = instance.state.mode == "customize" ? true : false;
    expect(result).toBe(true);

    // Change state to 'about' to simulate click
    instance.state.mode = "about";
    result = instance.state.mode == "about" ? true : false;
    expect(result).toBe(true);

    // Change state to 'choosingGame' to simulate click
    instance.state.mode = "choosingGame";
    result = instance.state.mode == "choosingGame" ? true : false;
    expect(result).toBe(true);
  });

  it('TEST #5 - CORRECT CONTENT APPEARS AFTER 3 MAIN PAGE OPTIONS ARE CLICKED', () => {
    // Create wrappers for components
    var customizeWrapper = mount(<Customize/>);
    var tabletopWrapper = mount(<Tabletop/>);

    // Create instances for wrappers
    var tabletopInstance = tabletopWrapper.instance();
    var customizeInstance = customizeWrapper.instance();

    // 1. Confirm correct cotent is showing for 'Play' tab
    tabletopWrapper.setState({ mode: 'choosingGame' });
    const foundChoices = tabletopWrapper.find('p');
    expect(foundChoices.first().text()).toBe("Choose A Game");
    expect(foundChoices.at(1).text()).toBe('War');
    expect(foundChoices.at(2).text()).toBe('Sandbox Mode');

    // 2. Confirm correct content is showing for 'Customize' tab
    const foundNumberOfImages = customizeWrapper.find('img').length;
    const desiredNumberOfImages = 8;
    const result = foundNumberOfImages === desiredNumberOfImages ? true : false;
    expect(result).toBe(true);

    // 3. Confirm correct content is showing for 'About' tab
    tabletopWrapper.setState({ mode: 'about' });
    const aboutMessage = tabletopWrapper.find('p');
    expect(aboutMessage.first().text()).toBe('About');
    expect(aboutMessage.at(aboutMessage.length - 1).text()).toBe('This is a web application developed by Braden Batman, Chase Grainger, and Matthew Heck. You can play premade games, or play around with a deck of cards and premade card layouts in Sandbox mode.');
  });

});
