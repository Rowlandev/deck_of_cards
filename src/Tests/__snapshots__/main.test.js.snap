// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DECK OF CARD TESTS TEST #1 - APPLICATION RENDERS 1`] = `
<div
  className="App"
>
  <div
    id="display"
  >
    <div
      className="main-menu"
    >
      <p
        id="title"
      >
        deck_of_cards
      </p>
      <p
        id="creators"
      >
        By: Braden Batman, Chase Grainger, and Matthew Heck
      </p>
      <p>
        <button
          className="button"
          onClick={[Function]}
        >
          Play
        </button>
      </p>
      <p>
        <button
          className="button"
          onClick={[Function]}
        >
          Customize
        </button>
      </p>
      <p>
        <button
          className="button"
          onClick={[Function]}
        >
          About
        </button>
      </p>
    </div>
  </div>
</div>
`;
